### PR TITLE
Implemented Dialog System 

### LINKED ISSUE #450: [TRACKER] NPC Editor features
# Editor
- [x] Behavior trees are editable
- [x] Behavior trees can be saved
- [x] Behavior trees can be opened (editor)
- [x] Behavior trees can be loaded (as states)
- [x] States that represent each BT 
- [x] The State machine can be saved
- [x] The State machine can be loaded (as the complete NPC behavior)
- [x] The State machine can be opened (editor)
- [x] Any other NPC information is stored in the SM file 
- [x] Sounds can be attached or referenced in editor
- [x] The editors can be accessed in the Godot Engine Editor
- [x] The whole system is an addon
- [x] Shortcuts to add different nodes in-editor
# Dialog System
- [x] Linear dialogs
- [x] Let the player answer
- [x] Dialogs based on clicking buttons
- [x] Dialogs based on code words
- [x] Branching dialogs
- [X] Have an editor for the dialogs
- [x] Dialogs can be loaded in their editor

### PR DESCRIPTION
**Pull request by yaelatletl**
Ready to test and merge, resolves #450

### CODE REVIEW

**Comment by Reousa** [comment]
How can I test this? @yaelatletl

**Comment by yaelatletl** [comment]
res://addons/joyeux_dialog_system/src/components/display/Dialog Display.tscn and play the scene, the test file it uses is res://test.mtalk

**Comment by Reousa** [comment]
Hitting skip isn't functional 
image

Otherwise, lgtm

File: addons/joyeux_dialog_system/src/components/display/Dialog Display.gd [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+func _on_Skip_pressed():
+	if PauseButton.pressed:
+		return
+	var temp = evaluate_node(next_node)
+	if temp is GDScriptFunctionState:
+		next_node = yield(temp, "completed")
+	elif temp is String:
+		next_node = temp
+	_on_Skip_pressed()
+
```
Comment (Reousa) [inline addons/joyeux_dialog_system/src/components/display/Dialog Display.gd:112]: This function causes the game to crash

**Review by Reousa — CHANGES_REQUESTED** [review]

File: addons/joyeux_dialog_system/src/components/display/Dialog Display.gd (hunk 1 above)
Comment (yaelatletl) [reply addons/joyeux_dialog_system/src/components/display/Dialog Display.gd]: latest comit 0b25130 solves this issue adding a yield

**Review by Reousa — APPROVED** [review]

**Pull request merged**